src/extension.ts

```diff
@@ -38,16 +38,21 @@ export function getFormattedText(
         CaseOptionEnum[<keyof typeof CaseOptionEnum>config.keywordCase];
     }
 
-    if (!formattingOptions.spaces && options.tabSize) {
-      // If spaces config not specified, use the FormattingOptions value from VSCode workspace
-      formattingOptions.spaces = Number(options.tabSize);
-    }
-    if (!formattingOptions.spaces && formattingOptions.tabs === undefined) {
-      // Neither spaces nor tabs option is configured, use insertSpaces to determine if we want to use tabs
-      formattingOptions.tabs = !options.insertSpaces;
+    if (!formattingOptions.spaces) {
+      if (options.tabSize) {
+        // If spaces config not specified, use the FormattingOptions value from VSCode workspace
+        formattingOptions.spaces = Number(options.tabSize);
+      }
+
+      if (formattingOptions.tabs === undefined) {
+        // Neither spaces nor tabs option is configured; use insertSpaces to determine if we want to use tabs
+        formattingOptions.tabs = options.insertSpaces === false;
```

Comment:
Main change here

Reply:
tbh I don't really know enough about vscode configuration to rate how exactly this impacts real world projects (e.g. what if `.editorconfig` has indent_style = tab etc). But looks fine to me.

Reply:
It's a little tricky but in general the VSCode `insertSpaces` setting can be controlled by extensions, VSCode settings, or as a fallback default, it gets set based on if the current file being edited uses tabs or spaces.  Editorconfig would set `insertSpaces` at runtime and override any VSCode level setting.